contracts/cw20-base/src/contract.rs

```diff
@@ -248,25 +268,29 @@ pub fn execute_transfer(
     }
 
     let rcpt_addr = deps.api.addr_validate(&recipient)?;
+    let sender_addr: &Addr = &info.sender;
 
-    BALANCES.update(
+    let sender_balance = BALANCES.update(
         deps.storage,
-        &info.sender,
+        sender_addr,
         |balance: Option<Uint128>| -> StdResult<_> {
             Ok(balance.unwrap_or_default().checked_sub(amount)?)
         },
     )?;
-    BALANCES.update(
+    
+    let receiver_balance = BALANCES.update(
         deps.storage,
         &rcpt_addr,
         |balance: Option<Uint128>| -> StdResult<_> { Ok(balance.unwrap_or_default() + amount) },
     )?;
 
     let res = Response::new()
         .add_attribute("action", "transfer")
-        .add_attribute("from", info.sender)
+        .add_attribute("from", sender_addr.as_str())
         .add_attribute("to", recipient)
-        .add_attribute("amount", amount);
+        .add_attribute("amount", amount)
+        .add_attribute("sender_balance", sender_balance)
```

Comment:
I think in general we tend to have events describe the action taken, not the outcome. That would get out of hand quickly.

Reply:
Thanks for all reviews
With the outcome, we will index it easier in offchain services; 
Assume that we have only from, to, amount, we won't know what's the target balance each account, we have to query the chain each time event happens, but it is not good

Reply:
Why is it bad? Queries don't cost gas.

Reply:
If we model this after the `send` messages in the Cosmos SDK bank module, there's minimal information there too:

https://docs.cosmos.network/master/modules/bank/section_04.html

I keep looking up the topic and it keeps occurring to me that events are not meant to provide full information for clients; they're meant to help block explorers and other tools index txs. [See here](https://docs.cosmos.network/master/core/events.html).

Reply:
> Why is it bad? Queries don't cost gas.
I agree that queries won't cost gas; but for some off-chain indexer/FE, querying a lot may cause nodes to be choked. So emit event on contract is a good choice for off-chain indexers, scaling off-chain services for querying is much easier than adding more nodes I think

Reply:
> https://docs.cosmos.network/master/modules/bank/section_04.html
The links are dead somehow xD

Anyway I understand that we can query the node, which is also good for user end; for some applications like what I'm working on, the outcome is better because we will know exactly what the result is after txs happened